src/utils/getWeatherStations.ts

```diff
@@ -32,23 +31,48 @@ export const adjustStationDataset = (dataset: IWeatherStation[]) => {
       });
     }
   }
+  return dataset;
 };
 
-export const findNearestActiveStation = async(targetLat: number, targetLong: number, fromDate: number | string,
-     toDate: number | string) => {
-  // TODO: filter out weather stations that are active
-  let nearestStation: IWeatherStation | null = null;
-  let minDistance = Number.MAX_VALUE;
-
-  for (const station of weatherStations) {
-    const distance = calculateDistance(targetLat, targetLong, station.latitude, station.longitude);
-    if (distance < minDistance) {
-      minDistance = distance;
-      nearestStation = station;
+export const findNearestActiveStations = async(targetLat: number, targetLong: number, fromDate: Date,
+     toDate: Date) => {
+  const adjustedStationDataset = adjustStationDataset(weatherStations as IWeatherStation[]);
+  const fromSecs = fromDate.getTime() / 1000;
+  const toSecs = toDate.getTime() / 1000;
+  const nearestStations: IStation[] = [];
+  console.log(adjustedStationDataset);
+
+  const insertStation = (station: IWeatherStation, distance: number) => {
+    const newStation = {station, distance};
+    // Insert the new station into the sorted array at the correct position
+    const index = nearestStations.findIndex(s => s.distance > distance);
+    if (index === -1) {
+      nearestStations.push(newStation);
+    } else {
+      nearestStations.splice(index, 0, newStation);
+    }
+  };
+
+  for (const station of adjustedStationDataset) {
+    let shouldInsert = false;
+    if (station.maxdate === "present") {  // If the station is still active (maxdate === "present") then we can use it
+      shouldInsert = true;
+    } else {  // If the station is not active, we need to check if it has data in the date range
+      const stationMinSecs = new Date(station.mindate).getTime() / 1000;
+      const stationMaxSecs = new Date(station.maxdate).getTime() / 1000;
+      if (stationMinSecs <= toSecs && stationMaxSecs >= fromSecs) {
```

Comment:
Just curious, is there a reason to convert the getTime() values to seconds? Or could we just directly compare the getTime() values?

Reply:
Was just legacy stuff